### ISSUE TITLE
cc_libraries unable to depend on objc_import

### ISSUE THREAD

**Issue by cpsauer** [bazelbuild/bazel#10740]
Hi awesome Bazel team,

Is there any way we could allow objc_imports to be direct deps of cc_libraries?  It'd be really handy when working with C or C++ libraries on Apple platforms.

For context, I'm trying to use a .a archive for iOS that has a C interface. It needs to be imported with objc_import, because (as you'd expect) it uses sdk_frameworks as dependencies.

Right now this fails with the error message
```
in deps attribute of cc_library rule <<omitted cc_library rule>>: objc_import rule <<omitted objc_import rule>> is misplaced here (expected cc_library, objc_library, cc_proto_library or cc_import) and <<omitted objc_import rule>> does not have mandatory providers: 'CcInfo'
```
This isn't ideal because it's pretty reasonable to have C/C++ interfaces (conditionally) used from C++ code on Apple platforms. One can (and I will) temporarily work around by doing multiple layers of rule wrappers, but that's fairly ugly. (The simple workaround is blocked by #10738).

Thanks,
Chris
(ex-Googler)

**Comment by oquenchil** [comment]
Hey Chris, you may be interested in this issue: https://github.com/bazelbuild/bazel/issues/10674

I think once the work there is done, it will reduce a lot of the pains you are having.

**Comment by cpsauer** [comment]
Hey @oquenchil, saw you closed this. Any chance you'd be willing to keep it open as a dependent of the issue you mentioned until the CC/objc interop stuff gets fixed? 

Definitely still feels like a bug/oversight (rather than support), since cc_libraries can depend on objc (as they should). Seems like the same should hold, even if the objc is precompiled.

**Comment by cpsauer** [comment]
@oquenchil ?

**Comment by cpsauer** [comment]
For anyone else looking, this was fixed in 4.0, along with a bunch of other objc_library issues bundled in #10674.
